mipengine/algorithms/logistic_regression.py

```diff
@@ -28,7 +28,7 @@ def run(algo_interface):
     global_run = algo_interface.run_udf_on_global_node
     get_table_schema = algo_interface.get_table_schema
 
-    classes = algo_interface.algorithm_parameters["classes"]
+    classes = Literal(value=algo_interface.algorithm_parameters["classes"])
```

Comment:
The way it was before you could pass any python `Object` you wanted as a  argument to the run_udf_...  methods. You still can, since the `value` parameter of the `Literal` class is of thype `Any`, but it makes it more structured to be able to pass only 2 types as arguments `INodeTable` or `Literal`, instead of `INodeTable` and any other python object, as it was before. 
Yes, it makes it a bit more verbose for the algo dev but I believe it makes more sense, @jassak you thoughts?

Reply:
I agree with @ThanKarab that declaring a `Literal` in the algorithm flow makes little sense. The word literal makes only sense in the context of the udf generator where these values are literally printed in the udf code, as opposed to passed as parameters. I think that it's OK to have any other type, except `INodeTable` passed as is and cast as a `Literal` implicitly.

On the other hand, I think that it would be much more beneficial to differentiate among the different subtypes of `INodeTable`. Currently the algorithm executor is only able to send a single table type in the node/udfgenerator. However, the udfgenerator differentiates among different subtypes (tensors, relations, transfers, states, etc.) and is forced to employ heuristics to infer the table subtype. Udfs declare the returned type, so it's easy to have this information in the algorithm flow. That would eliminate the need for these heuristics.

Reply:
@jassak I don't think it's a good idea to do the 2nd part that you suggest.

The udfgenerator is completelly separate from the algorithm executor and the objects should not be mixed together. This would mean that the udfgenerator types would now be all over the node/controller, instead of the necessary (tablename) information.
The MOST important thing though is that the node should NOT trust information coming from the controller so even if the controller has information about the type of the input, the node should "figure" it out from the table. This is especially important on "state" types, etc.
(Maybe I am a bit overdramatic on the secutiry but I really think we shouldn't mix these things together, we will open the aeolian bag. Overly dramatic again...)

Reply:
removed the Literal type